### ISSUE TITLE
coding errors in actions don't raise any error

### ISSUE THREAD

**Issue by disimone** [RasaHQ/rasa_core#451]
**Rasa Core version**:
Master (commit 241979ca4fbe22083aec7b1b6c37810cc0ce2e6c)

**Python version**: 
3.5.2
**Operating system** (windows, osx, ...):
linux
**Issue**:

Hi, I noticed that coding errors in the actions are not reported anywhere, when running the bot.

Even macroscopic errors like using an undefined variable are just silenced.

It would seem like somewhere under the hood you are running the action code with a very wide exception catching policy. I don't think this should be happening. Please just catch expections that are generated by rasa, and pass the other ones up so we can deal with them ourselves.

Thanks for your help,

Andrea.

**Comment by akelad** [comment]
If you run the bot with debug mode enabled (so pass --debug to the rasa_core.run script) all the exceptions are shown

**Comment by tmbo** [comment]
Yes we are catching all exceptions there. But you are right, we should probably only do that in production mode or something like that, because otherwise the bot might be unreachable because of a small bug.

**Comment by disimone** [comment]
I must admit I am not sure what the best thing here is. I see how you do not want the bot to crash on a conversation just because of an exception.

I think the choice to keep the bot running is the best one, but some uttering should warn the user that something went wrong. in my case the exception was preventing some data to be retrieved from a db, which made the rest of the conversation completely meaningless.

You could maybe introduce a RasaCoreError and a RasaCoreCritical exception that users can raise in their code whenever some crucial part fails. If it is a simple error, you continue the conversation. with the most severe exception continuing the conversation makes no sense. then you can handle those two on your side with some templated messages, plus conversation terminate/restart if needed.

jm2c,

Andrea.
